mono/metadata/file-io.c

```diff
@@ -489,8 +489,11 @@ ves_icall_System_IO_MonoIO_FindFirst (MonoString *path,
 	ifh = g_new (IncrementalFind, 1);
 	ifh->find_handle = find_handle;
 	ifh->utf8_path = mono_string_to_utf8_checked (path, &error);
-	if (mono_error_set_pending_exception (&error))
+	if (mono_error_set_pending_exception (&error)) {
+		FindClose (find_handle);
```

Comment:
Yeah, I don't know the details either, I just saw it when comparing what `ves_icall_System_IO_MonoIO_FindClose` does :smile:

@ludovic-henry is skipping those macros fine in this case?